core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -1829,7 +1895,50 @@ private[spark] class BlockManager(
     data.dispose()
   }
 
+  /**
+   * Class to handle block manager decommissioning retries
+   * It creates a Thread to retry offloading all RDD cache blocks
+   */
+  private class BlockManagerDecommissionManager {
+    @volatile private var stopped = false
+    private val cacheReplicationThread = new Thread {
+      override def run(): Unit = {
+        while (blockManagerDecommissioning && !stopped) {
+          try {
+            logDebug(s"Attempting to replicate all cached RDD blocks")
+            offloadRddCacheBlocks()
+            logInfo(s"Attempt to replicate all cached blocks done")
+            val sleepInterval = if (Utils.isTesting) 100 else 30000
+            Thread.sleep(sleepInterval)
+          } catch {
+            case _: InterruptedException =>
+            // no-op
+            case NonFatal(e) =>
+              logError("Error occurred while trying to " +
+                "replicate cached RDD blocks for block manager decommissioning", e)
+          }
+        }
+      }
+    }
+    cacheReplicationThread.setDaemon(true)
+    cacheReplicationThread.setName("cache-replication-thread")
+
+    def start(): Unit = {
```

Comment:
Shall we have a `logInfo` to match with line 1933, `logInfo("Stopping cache replication thread")`?

Reply:
done.

Reply:
added logInfo.